filter.cc

```diff
@@ -34,26 +35,32 @@ Filter::Filter() :
   dataLen(-1),
   domainList(nullptr),
   host(nullptr),
-  domainCount(0),
-  antiDomainCount(0),
-  hostLen(-1) {
+  hostLen(-1),
+  domains(nullptr),
+  antiDomains(nullptr),
+  domainsParsed(false) {
 }
 
 Filter::~Filter() {
-  if (borrowed_data) {
-    return;
-  }
-  if (data) {
-    delete[] data;
+  if (domains) {
+    delete domains;
   }
-  if (ruleDefinition) {
-    delete[] ruleDefinition;
-  }
-  if (domainList) {
-    delete[] domainList;
+  if (antiDomains) {
+    delete antiDomains;
   }
-  if (host) {
-    delete[] host;
+  if (!borrowed_data) {
```

Comment:
This is just a refactoring of an early return to not have an early return.  Code in this block is the same as before.